### ISSUE TITLE
Supporting container SAS URLs for Azure Blob Storage

### ISSUE THREAD

**Issue by cbare** [RaRe-Technologies/smart_open#651]
#### Supporting container SAS URLs for Azure Blob Storage

Thanks for making a very handy library!

I'm trying to compress batches of files into large archives and stream that to Azure Blob Storage. Based on a quick test, this code seems to work.

```python
import smart_open
from glob import iglob

transport_params = {
    'client': BlobServiceClient.from_connection_string(blob_connect_string),
}

with smart_open.open("azure://data/ziptest/batch.zip", mode="wb", transport_params=transport_params) as blob_stream:
    with zipfile.ZipFile(blob_stream, "w") as zf:
        for source_file in iglob(source_path+"/*"):
            with open(source_file, "rb") as sf:
                data = sf.read()
            zf.writestr(os.path.basename(source_file), data)
```

Based on least-privilege, I'd like to only give write permission to a specific container. Maybe something like:

```python
transport_params = {
    'container_client': ContainerClient.from_container_url(container_sas_url),
}
```

I'm wondering if you think something like that would be worth supporting? How difficult would it be to add? Or is there already support and I just haven't found it, yet? Thanks!

#### Versions

Linux-5.10.16.3-microsoft-standard-WSL2-x86_64-with-glibc2.10
Python 3.8.5 (default, Sep  4 2020, 07:30:14) 
[GCC 7.3.0]
smart_open 5.2.1

**Comment by cbare** [comment]
Ah flake8, my old friend. Apparently, flake8 takes offense at long lines caused by the type annotations. I added a type alias to resolve that. As this introduces the first import from typing, does that impact support for older versions of Python?

**Comment by cbare** [comment]
I had a go at implementing this. Please let me know what you think.

**Comment by mpenkov** [comment]
Fixed by #652
